model/build/build.go

```diff
@@ -181,12 +181,12 @@ func (b *Build) UpdateMakespans(predictedMakespan, actualMakespan time.Duration)
 	)
 }
 
-// TryMarkBuildStarted attempts to mark a b as started if it
+// TryMarkStarted attempts to mark a build as started if it
 // isn't already marked as such
 func TryMarkStarted(buildId string, startTime time.Time) error {
 	selector := bson.M{
 		IdKey:     buildId,
-		StatusKey: evergreen.BuildCreated,
+		StatusKey: bson.M{"$ne": evergreen.BuildStarted},
```

Comment:
It seems a little bit strange to allow finished statuses to be marked as started. Is this to have the correct status if you restart a task in a finished build?

Reply:
That's what I was thinking, although I'm not sure if this status would already have been fixed because this is only called by the agent, and the statuses might have been fixed already by then, but it didn't seem harmful to allow for it just in case.